include/seqan3/alignment/pairwise/policy/unbanded_score_trace_dp_matrix_policy.hpp

```diff
@@ -0,0 +1,114 @@
+                                                                           score_allocator_t,
+                                                                           trace_allocator_t>,
+                                     score_allocator_t>
+{
+private:
+
+    //!\brief The base type
+    using base_t = unbanded_dp_matrix_policy<unbanded_score_trace_dp_matrix_policy<derived_t,
+                                                                                   score_allocator_t,
+                                                                                   trace_allocator_t>,
+                                             score_allocator_t>;
+
+    //!\brief Befriends the derived class to grant it access to the private members.
+    friend derived_t;
+    //!\brief Make dimension_first_range visible in this class.
+    using base_t::dimension_first_range;
+    //!\brief Make dimension_second_range visible in this class.
+    using base_t::dimension_second_range;
+    //!\brief Make score_matrix visible in this class.
+    using base_t::score_matrix;
+
+    /*!\name Member types
+     * \{
+     */
+    //!\brief The underlying cell type of the dynamic programming matrix.
+    //!\brief The underlying cell type of the scoring matrix.
+    using cell_type = typename score_allocator_t::value_type;
+    //!\brief The underlying cell type of the trace matrix.
+    using trace_type = typename trace_allocator_t::value_type;
+    //!\brief The type of the score matrix.
+    using score_matrix_type = std::vector<cell_type, score_allocator_t>;
+    //!\brief The type of the trace matrix.
+    using trace_matrix_type = std::vector<trace_type, trace_allocator_t>;
+    //!\}
+
+    /*!\name Constructor, destructor and assignment
+     * \{
+     */
+    constexpr unbanded_score_trace_dp_matrix_policy() = default;
+    constexpr unbanded_score_trace_dp_matrix_policy(unbanded_score_trace_dp_matrix_policy const &) = default;
+    constexpr unbanded_score_trace_dp_matrix_policy(unbanded_score_trace_dp_matrix_policy &&) = default;
+    constexpr unbanded_score_trace_dp_matrix_policy & operator=(unbanded_score_trace_dp_matrix_policy const &) = default;
+    constexpr unbanded_score_trace_dp_matrix_policy & operator=(unbanded_score_trace_dp_matrix_policy &&) = default;
+    ~unbanded_score_trace_dp_matrix_policy() = default;
+    //!}
+
+    /*!\brief Allocates the memory for the dynamic programming matrix given the two sequences.
+     * \tparam first_range_t   The type of the first sequence (or packed sequences).
+     * \tparam second_range_t  The type of the second sequence (or packed sequences).
+     * \param[in] first_range  The first sequence (or packed sequences).
+     * \param[in] second_range The second sequence (or packed sequences).
+     */
+    template <typename first_range_t, typename second_range_t>
+    constexpr void allocate_matrix(first_range_t && first_range, second_range_t && second_range)
```

Comment:
question: will you ever receive rvalues here?

Reply:
This will be refactored anyway. as I should not use forwarding references here, since we cannot steal the memory anyway. But I make this in a clean PR.

Reply:
maybe you should make a list of follow up issues then...

Reply:
I included this now in this PR as well.